packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1607,6 +1615,9 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
   ScrollController? _internalScrollController;
   ScrollController get _scrollController => widget.scrollController ?? (_internalScrollController ??= ScrollController());
 
+  UndoHistoryController? _undoController;
```

Comment:
Is this never assigned or did I miss that somewhere?

Should UndoHistory create and manage the UndoHistoryController if it's not provided?  Then we could also get rid of _effectiveUndoController.

Reply:
It does get assigned at the end of line 1619...but...to your other point...I think you might be right that we could let UndoHistory manage that case, since we aren't doing anything here except making sure there is an UndoHistoryController